cmd/dellhw_exporter/dellhw_exporter.go

```diff
@@ -62,8 +65,23 @@ var (
 type DellHWCollector struct {
 	lastCollectTime time.Time
 	collectors      map[string]collector.Collector
+
+	// Cache related
+	cachingEnabled bool
+	cacheDuration  time.Duration
+	cache          []prometheus.Metric
+	cacheMutex     sync.Mutex
 }
 
+func NewDellHWCollector(collectors map[string]collector.Collector, cachingEnabled bool, cacheDurationSeconds int64) *DellHWCollector {
+	return &DellHWCollector{
+		cache:           make([]prometheus.Metric, 0),
+		lastCollectTime: time.Unix(0, 0),
+		collectors:      collectors,
+		cachingEnabled:  cachingEnabled,
+		cacheDuration:   time.Duration(cacheDurationSeconds) * time.Second,
+	}
+}
 func init() {
```

Comment:
```suggestion
}

func init() {
```

Please add an empty line between the function.

Reply:
Good catch, thanks!
Unfortunately `go fmt` did not caught it.